### PR TITLE
Update FrontDoor WAF API with new preview features (2019-03-01)

### PR DESCRIPTION
**Pull request by eschwabe**
### Review Status
The new WAF API version has already been reviewed in the private repo ([**PR 690**](https://github.com/Azure/azure-rest-api-specs-pr/pull/690)).

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5527 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5527/commits/0a5332d9420ea785a0d5e1ec44436d6396b46353

**Comment by dsgouda** [comment]
Got some other deliverables, reassigning this one.

File: specification/frontdoor/resource-manager/Microsoft.Network/preview/2019-03-01-preview/examples/WafListPolicies.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "parameters": {
-        "api-version": "2018-08-01",
+        "api-version": "2019-03-01",
```
Comment (eschwabe) [inline specification/frontdoor/resource-manager/Microsoft.Network/preview/2019-03-01-preview/examples/WafListPolicies.json]: API changes have already been reviewed in the private repo ([**PR 690**](https://github.com/Azure/azure-rest-api-specs-pr/pull/690)).

**Comment by eschwabe** [comment]
API changes have already been reviewed in the private repo ([**PR 690**](https://github.com/Azure/azure-rest-api-specs-pr/pull/690)).

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**